Remove Section#has_ever_been_published?

chrisroos opened the pull request:
This follows on from PR #1159 and replaces the final use of `Section#has_ever_been_published?` before then removing the method.

floehopper commented on spec/view_adapters/section_view_adapter_spec.rb:
```diff
@@ -0,0 +1,37 @@
+      expect(subject).to_not be_first_edition
+      expect(subject.accepts_minor_updates?).to be(true)
+    end
+  end
```
Minor: Is it worth including the final logical combination, i.e. not first edition and published?

chrisroos replied:
I've updated the first commit to include this test.

chrisroos commented:
Thanks for the review, @floehopper. I've modified the first commit based on your feedback, rebased on master and force pushed in preparation for merging.
